File: pythonfiles/triangle.py
INVALID = -1
SCALENE = 0
ISOSCELES = 1
EQUILATERAL = 2


def triangle_identifier(a, b, c):

    is_a_triangle = None

    # Is A Triangle?
    if (a < b + c) and (b < a + c) and (c < a + b):
        is_a_triangle = True
    else:
        is_a_triangle = False

    # Determine Triangle Type
    triangle_type = INVALID
    if is_a_triangle:
        if (a == b) and (b == c):
            triangle_type = EQUILATERAL
        elif (a != b) and (a != c) and (b != c):
            triangle_type = SCALENE
        else:
            triangle_type = ISOSCELES
    else:
        triangle_type = INVALID

    return triangle_type

File: src/main/java/assignment01/Triangle.java
package assignment01;

public class Triangle {

    public static final int INVALID = -1;
    public static final int SCALENE = 0;
    public static final int ISOSCELES = 1;
    public static final int EQUILATERAL = 2;

    public static int triangleIdentifier(int a, int b, int c) {

        boolean isATriangle;

        // Is A Triangle?
        if ((a < b + c) && (b < a + c) && (c < a + b))
            isATriangle = true;
        else
            isATriangle = false;

        // Determine Triangle Type
        int triangleType = INVALID;
        if (isATriangle) {
            if ((a == b) && (b == c))
                triangleType = EQUILATERAL;
            else if ((a != b) && (a != c) && (b != c))
                triangleType = SCALENE;
            else
                triangleType = ISOSCELES;
        } else
            triangleType = INVALID;

        return triangleType;
    }
}